expression/builtin_compare.go

```diff
@@ -407,12 +407,15 @@ func resolveType4Extremum(args []Expression) types.EvalType {
 	if aggType.EvalType().IsStringKind() {
 		for i := range args {
 			item := args[i].GetType()
-			if types.IsTemporalWithDate(item.Tp) {
-				temporalItem = item
+			// Find the temporal value in the arguments but prefer DateTime value.
+			if types.IsTemporal(item.Tp) {
+				if temporalItem == nil || temporalItem.Tp == mysql.TypeDatetime {
```

Comment:
// Find the temporal value in the arguments but **prefer** DateTime value?
Should the condition be `temporalItem == nil || temporalItem.Tp != mysql.TypeDatetime`?